src/gui/transferlistwidget.cpp

```diff
@@ -1179,7 +1179,8 @@ void TransferListWidget::currentChanged(const QModelIndex &current, const QModel
     {
         torrent = m_listModel->torrentHandle(mapToSource(current));
         // Scroll Fix
-        scrollTo(current);
+        // Delay the call to scrollTo until control is passed back to the event queue
+        QTimer::singleShot(0, [this, &current] { scrollTo(current); });
```

Comment:
Please use this function instead: https://doc.qt.io/qt-5/qmetaobject.html#invokeMethod-4 with `(this, lambda, Qt::QueuedConnection)`.
`QTimer::singleShot` is sort of a hack IMO.

Reply:
> `QTimer::singleShot` is sort of a hack IMO.

I suppose you mean zero timeout when talking about "hack". There is nothing wrong in `QTimer::singleShot` itself.

Reply:
You shouldn't capture temporary by reference if lambda is supposed to be used asynchronously.

Reply:
Changed as suggested.

Reply:
> > I suppose you mean zero timeout when talking about "hack".
> 
> Exactly.

I wouldn't call it a "hack", it's always been just a "feature" with documented behavior. But `QMetaObject::invokeMethod()` makes things more explicit, so I agree with you about preferring it.